packages/aardvark-react/src/default_landing_page.tsx

```diff
@@ -150,14 +207,33 @@ export class DefaultLanding extends React.Component<DefaultLandingProps, Default
 		}
 
 		return <>
-			<div className="LandingButton" onClick={ this.onAddFavorite }>Add to Favorites</div>
-			<div className="LandingButton" onClick={ this.onRemoveFavorite }>Remove from Favorites</div>
+			{ this.renderMutallyExclusiveButtons( 
+				this.onAddFavorite, 
+				this.onRemoveFavorite, 
+				"Favorites", 
+				this.state.hostSettings?.favorited ) }
+
+			{ this.renderMutallyExclusiveButtons( 
+				this.onSetAutoLaunch, 
+				this.onRemoveAutoLaunch, 
+				"Auto Launch", 
+				this.state.hostSettings?.autoLaunchEnabled ) }
+
 			<div className="LandingButton" onClick={ this.onStartGadget }>Start Gadget</div>
 			{ this.state.addResult &&
 				<div style={ { fontSize: "medium" }}>{ this.state.addResult }</div> }
-			</>;
+		</>;
 	}
 
+	private renderMutallyExclusiveButtons( enableCallback: () => void, disableCallback: () => void, description: string, testValue?: boolean )
+	{
+		return <>
+			{ (testValue === undefined || !testValue) && 
+				<div className="LandingButton" onClick={ enableCallback }>Add to { description }</div> }
+			{ (testValue === undefined || testValue) && 
+				<div className="LandingButton" onClick={ disableCallback }>Remove from { description }</div> }
```

Comment:
Why show both buttons if this is undefined? Seems like undefined would just be a default (unfavored, and not autostart)